### ISSUE TITLE
[Feature]  Grammar/structured output support

### ISSUE THREAD

**Issue by nidhoggr-nil** [InternLM/lmdeploy#1614]
### Motivation

Many agent and llm frameworks are starting to use structured output to make integration and parsing output work consistently.

I could not find any documentation if turbomind supports this or if it is in the works, but it seems very relevant.

Something like [lm-format-enforcer](https://github.com/noamgat/lm-format-enforcer) would be very nice to have.

This should then be made to support tool calls via the openai API, so that turbomind becomes openai REST compatible with any framework.

**Comment by suc16** [comment]
#1664 instructor

**Comment by lvhan028** [comment]
@AllentDan is working on this feature

**Comment by QwertyJack** [comment]
Should be working now after #1856
